### PR TITLE
Typecast bool to integer and vice versa

### PR DESCRIPTION
**Pull request by Zainab-Saad**
This PR is follow up of issue #824 
- Implement typecasting boolean to integer and vice versa
- Add regression tests for the implementation

### CODE REVIEW

File: src/backend/parser/cypher_expr.c [hunk 1]
```diff
@@ -955,6 +956,10 @@ static Node *transform_cypher_typecast(cypher_parsestate *cpstate,
     {
         fname = lappend(fname, makeString(FUNC_AGTYPE_TYPECAST_PG_BIGINT));
     }
+    else if ((pg_strcasecmp(ctypecast->typecast, "bool") == 0 || pg_strcasecmp(ctypecast->typecast, "boolean") == 0))
```
Comment (jrgemignani) [inline src/backend/parser/cypher_expr.c:959]: Please make sure to conform to our ~80 character per line coding standard.

File: src/backend/utils/adt/agtype.c [hunk 2]
```diff
@@ -2575,7 +2575,7 @@ Datum agtype_to_bool(PG_FUNCTION_ARGS)
     agtype_value agtv;
 
     if (!agtype_extract_scalar(&agtype_in->root, &agtv) ||
-        agtv.type != AGTV_BOOL)
+        (agtv.type != AGTV_BOOL && agtv.type != AGTV_INTEGER))
```
Comment (jrgemignani) [inline src/backend/utils/adt/agtype.c:2578]: All ifs need to have {} even for 1 line of code. If you are modifying old code that doesn't follow that standard, please update it. No need to update everything, just the contents of the function.

File: src/backend/utils/adt/agtype.c [hunk 3]
```diff
@@ -2654,7 +2654,8 @@ Datum agtype_to_int4(PG_FUNCTION_ARGS)
         (agtv.type != AGTV_FLOAT &&
          agtv.type != AGTV_INTEGER &&
          agtv.type != AGTV_NUMERIC &&
-         agtv.type != AGTV_STRING))
+         agtv.type != AGTV_STRING &&
+         agtv.type != AGTV_BOOL))
```
Comment (jrgemignani) [inline src/backend/utils/adt/agtype.c:2658]: All ifs need to have {} even for 1 line of code. If you are modifying old code that doesn't follow that standard, please update it. No need to update everything, just the contents of the function.

**Review by jrgemignani — CHANGES_REQUESTED** [review]

**Comment by jrgemignani** [comment]
@Zainab-Saad Please address my review comments when you have an opportunity. 

Also, good job :)

File: src/backend/utils/adt/agtype.c (hunk 3 above)
Comment (Zainab-Saad) [reply src/backend/utils/adt/agtype.c:2658]: Since none of the 'if', 'else if' conditions have {} for 1 line of code inside these functions, do I add to all the conditions that I have not modified or to only the ones I have changed?

Comment (jrgemignani) [reply src/backend/utils/adt/agtype.c:2658]: If they are in the function that you are working on, that would be great. We're slowly trying to update these as we come across them. I wouldn't do more than just the function(s) that you are working on.

File: src/backend/parser/cypher_expr.c (hunk 1 above)
Comment (Zainab-Saad) [reply src/backend/parser/cypher_expr.c]: Done

**Comment by Zainab-Saad** [comment]
@jrgemignani I have made the requested changes, kindly take a look on it. Thanks

**Review by jrgemignani — APPROVED** [review]

**Comment by jrgemignani** [comment]
@Zainab-Saad Please squash your commit's extended description

**Comment by Zainab-Saad** [comment]
@jrgemignani For clarity, do you mean to squash the resolve merge conflict commits and add description for each commit after that?

**Comment by jrgemignani** [comment]
@Zainab-Saad The detailed commit description is just for details on what the changes are. You don't need any merge stuff in it.

**Comment by jrgemignani** [comment]
@Zainab-Saad There should be a document/blog by @rafsun42 that explains how to do this.

**Comment by Zainab-Saad** [comment]
@jrgemignani I have read the article. Actually the issue I am facing is that after resolving the merge conflict, my rebase interface does not allow squashing or deleting commit.

**Comment by jrgemignani** [comment]
@Zainab-Saad I believe that you need to do a force to update your PR. I know, force commands are generally not a good idea. However, for updating a PR of yours, they are okay.

**Comment by Zainab-Saad** [comment]
@jrgemignani Apologies if I was not clear before, the issue is that I used some cherry-pick command to resolve merge conflict and ended up with these so many cluttered commits that do nothing (don't resolve conflict :) ). I ended up resolving the conflict on github web rather than from my local system. So squashing or deleting will definitely cause merge conflicts to arise again. And I can not rebase properly because it results in empty commits, (--alow-empty) is not helping right now.
Yes, you are definitely right about force pushing but I would go on force pushing once rebase is done properly.
In short, I am facing issues while rebasing

**Comment by Zainab-Saad** [comment]
@jrgemignani Okay, I will close this PR.
